### PR TITLE
Flink 1.17: Use awaitility instead of Thread.sleep()

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

File: flink/v1.17/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceFailover.java [hunk 1]
```diff
@@ -98,9 +98,9 @@ protected List<Record> generateRecords(int numRecords, long seed) {
     return RandomGenericData.generate(schema(), numRecords, seed);
   }
 
-  protected void assertRecords(
-      Table table, List<Record> expectedRecords, Duration interval, int maxCount) throws Exception {
-    SimpleDataUtil.assertTableRecords(table, expectedRecords, interval, maxCount);
+  protected void assertRecords(Table table, List<Record> expectedRecords, Duration interval)
```
Comment (nk1506) [inline flink/v1.17/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceFailover.java:101]: do we need this method? can't we replace with 
`SimpleDataUtil.assertTableRecords(table, expectedRecords, interval);` ?

File: flink/v1.17/flink/src/test/java/org/apache/iceberg/flink/source/TestStreamingMonitorFunction.java [hunk 2]
```diff
@@ -111,14 +113,19 @@ public void testConsumeWithoutStartSnapshotId() throws Exception {
       TestSourceContext sourceContext = new TestSourceContext(latch);
       runSourceFunctionInTask(sourceContext, function);
 
-      Assert.assertTrue(
-          "Should have expected elements.", latch.await(WAIT_TIME_MILLIS, TimeUnit.MILLISECONDS));
-      Thread.sleep(1000L);
+      Awaitility.await()
```
Comment (nk1506) [inline flink/v1.17/flink/src/test/java/org/apache/iceberg/flink/source/TestStreamingMonitorFunction.java:116]: IMO, we should move this common part at one place.

File: flink/v1.17/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceFailover.java (hunk 1 above)
Comment (nastra) [reply flink/v1.17/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceFailover.java:101]: that's a valid point, I'm not sure why this method existed in the first place. I've removed it

**Review by nk1506 — APPROVED** [review]

**Comment by nastra** [comment]
thanks for reviewing @amogh-jahagirdar and @nk1506. @nk1506 would you be interested in backporting this to older Flink versions?

**Comment by nk1506** [comment]
Yeah sure @nastra . I will create PR with other versions.

**Pull request merged**